### PR TITLE
Return subcommand suggestions. Fixes #472

### LINKED ISSUE #472: Tab complete - SimpleDispatcher
In SimpleDispatcher.java, we are getting a list of tab complete options but never actually adding them to the result list. We should change line [341](https://github.com/SpongePowered/SpongeAPI/blob/master/src/main/java/org/spongepowered/api/util/command/dispatcher/SimpleDispatcher.java#L341) from:
```java
mapping.get().getCallable().call(source, parts.length > 1 ? parts[1] : "", Collections.unmodifiableList(passedParents)));
```
to:
```java
suggestions.addAll(mapping.get().getCallable().call(source, parts.length > 1 ? parts[1] : "", Collections.unmodifiableList(passedParents));
```

### PR DESCRIPTION
**Pull request by boformer**
This fixes the issue that the ``SimpleDispatcher`` does not return TAB complete suggestions that were generated by a subcommand.

### CODE REVIEW

File: src/main/java/org/spongepowered/api/util/command/dispatcher/SimpleDispatcher.java [hunk 1]
```diff
@@ -338,10 +339,25 @@ public synchronized boolean testPermission(CommandSource source) {
...
+                            //Join the alias and the suggestion, then add it to the suggestion list
+                            suggestions.add(joiner.join(alias, suggestion));
+                        }
+                        
+                        break;
+                    }
+                }
             }
         }
-
```
Comment (me4502) [inline src/main/java/org/spongepowered/api/util/command/dispatcher/SimpleDispatcher.java]: This change here isn't necessary.

**Comment by boformer** [comment]
This could also be an issue: https://github.com/SpongePowered/Sponge/issues/192
With this kind of TAB completion, there are no permission checks. I will fix it today.

File: src/main/java/org/spongepowered/api/util/command/dispatcher/SimpleDispatcher.java [hunk 2]
```diff
@@ -327,6 +328,9 @@ public synchronized boolean testPermission(CommandSource source) {
 
             synchronized (this) {
                 for (CommandMapping mapping : this.commands.values()) {
+                    // Skip commands the source is not permitted to call
+                    if(mapping.getCallable().testPermission(source)) continue;
```
Comment (ST-DDT) [inline src/main/java/org/spongepowered/api/util/command/dispatcher/SimpleDispatcher.java]: Is there a NOT missing here?
The comment above hints that.

Comment (Minecrell) [inline src/main/java/org/spongepowered/api/util/command/dispatcher/SimpleDispatcher.java]: There is also a space missng after the if, maybe reformat your code to fix it

File: src/main/java/org/spongepowered/api/util/command/dispatcher/SimpleDispatcher.java [hunk 3]
```diff
@@ -337,8 +341,25 @@ public synchronized boolean testPermission(CommandSource source) {
...
+            // Check if subcommand exists and source is permitted to call it
+            if (mapping.isPresent() && mapping.get().getCallable().testPermission(source)) {
+                // A list of suggestions delivered by the subcommand
+                List<String> subSuggestions = mapping.get().getCallable().getSuggestions(source, parts.length > 1 ? parts[1] : "");
+
+                // parts[0] is the alias how it was entered by the source, e.g. "sENd"
+                // Find the alias how it was registered, e.g. "Send"
+                for (String alias : mapping.get().getAllAliases()) {
+                    if (alias.equalsIgnoreCase(parts[0])) {
+                        Joiner joiner = Joiner.on(' ');
```
Comment (Minecrell) [inline src/main/java/org/spongepowered/api/util/command/dispatcher/SimpleDispatcher.java]: There is no need to initialize a new `Joiner` on every iteration, make it a constant in the class

**Comment by me4502** [comment]
This is looking quite nice

**Comment by Zidane** [comment]
@SpongePowered/developers Feedback please.

**Comment by me4502** [comment]
I feel this is mergable.

**Pull request closed without merging**